java/java/src/main/java/org/signal/zkgroup/groups/ClientZkGroupCipher.java

```diff
@@ -62,35 +62,13 @@ public ProfileKey decryptProfileKey(ProfileKeyCiphertext profileKeyCiphertext, U
   }
 
   public byte[] encryptBlob(SecureRandom secureRandom, byte[] plaintext) throws VerificationFailedException {
-
-    byte[] paddedPlaintext = new byte[plaintext.length + 4];
-    System.arraycopy(plaintext, 0, paddedPlaintext, 4, plaintext.length);
-
     byte[] random      = new byte[RANDOM_LENGTH];
-
     secureRandom.nextBytes(random);
-
-    return Native.GroupSecretParams_EncryptBlobDeterministic(groupSecretParams.getInternalContentsForJNI(), random, paddedPlaintext);
+    return Native.GroupSecretParams_EncryptBlobWithPaddingDeterministic(groupSecretParams.getInternalContentsForJNI(), random, plaintext, 0);
```

Comment:
Is 0 here the padding length (so no padding) or byte value to pad with? Seems like it should be 4?

Reply:
It's the padding length. Specifically, the way this encoding works is that it has a variable number of *trailing* padding bytes, and the number of trailing padding bytes is encoded as a big-endian 32-bit integer prepended to the actual message. The "4" in the old code is just `sizeof(u32)`.